File: spec/rails_app/app/components/client_config.rb
class ClientConfig < Netzke::Base
  def configure(c)
    super
    c.layout = :hbox
    c.defaults = { height: "100%" }
    c.some_option = client_config.some_option
    c.bbar = [:show_option_one, :show_option_two]

    child_config = { klass: HelloUser, flex: 1 }
    c.items = [child_config.merge(item_id: 'left'), child_config.merge(item_id: 'right')]
  end

  endpoint :request_some_option do
    config.some_option
  end

  action :show_option_one do |c|
    c.handler = :handle_show_option
    c.option = 'One'
  end

  action :show_option_two do |c|
    c.handler = :handle_show_option
    c.option = 'Two'
  end
end

File: spec/rails_app/app/components/client_config/client/client_config.js
{
  onShowOption: function(action) {
    this.serverConfig.some_option = action.option;
    this.server.requestSomeOption(null, function(res){this.setTitle(res);});

    var left = this.netzkeGetComponent('left');
    var right = this.netzkeGetComponent('right');

    left.serverConfig = {user_name: action.option + " Left"};
    right.serverConfig = {user_name: action.option + " Right"};
    left.onPingServer();
    right.onPingServer();
  }
}
